src/probabilities_estimators/histograms/rectangular_binning.jl

```diff
@@ -206,19 +179,89 @@ function RectangularBinEncoding(::AbstractDataset{D, T},
+# Then, the dataset doesn't need to be provided.
+function RectangularBinEncoding(b::FixedRectangularBinning{<:NTuple}, n_eps = 1)
+    ϵmin, ϵmax = b.ϵmin, b.ϵmax
+    D = length(ϵmin)
+    mini = SVector{D, Float64}(ϵmin)
+    maxi = SVector{D, Float64}(ϵmax)
+    edgelengths_nonadjusted = @. (maxi .- mini) / b.N
+    edgelengths = nextfloat.(edgelengths_nonadjusted, n_eps)
+    RectangularBinEncoding(b, mini, edgelengths)
+end
+
+##################################################################
+# Outcomes / total outcomes
+##################################################################
+const RBE = RectangularBinEncoding
+const RB = RectangularBinning
+const NONDEDUCIBLE{T} = Union{
+    RB{T},
+    RBE{RB{T}}
+    } where T <: Union{Q, Vector{Q}} where Q <: AbstractFloat
+
+function total_outcomes(x, symbolization::NONDEDUCIBLE)
+    msg = "total_outcomes can't be deduced from $NONDEDUCIBLE"
+    throw(ArgumentError(msg))
+end
 
+# multiple-axis bins don't make sense for univariate input.
+function total_outcomes(::AbstractVector,symbolization::RBE{RB{Vector{Int}}})
+    msg = "total_outcomes is ambiguous for Vectors with RectangularBinning{Vector{Int}}"
+    throw(ArgumentError(msg))
+end
+total_outcomes(::AbstractVector,symbolization::RBE{RB{Int}}) =
+    symbolization.binning.ϵ
+total_outcomes(::AbstractDataset{D}, symbolization::RBE{RB{Int}}) where {D} =
+    symbolization.binning.ϵ^D
+total_outcomes(::AbstractDataset{D}, symbolization::RBE{RB{Vector{Int}}}) where {D} =
+    prod(symbolization.binning.ϵ)
 
 # When the grid is fixed by the user, we can always deduce the total number of bins,
 # even just from the binning itself - symbolization info not needed.
-const FRB = FixedRectangularBinning
-total_outcomes(b::FRB) = b.N
-total_outcomes(::AbstractVector, b::FRB) = b.N
-total_outcomes(::AbstractDataset{D, T}, b::FRB) where {D, T} = b.N^D
-total_outcomes(symbolization::RBE{B, T}) where {B <: FRB, T <: Number} =
-    symbolization.binning.N
-total_outcomes(symbolization::RBE{B, T}) where {B <: FRB, T <: SVector{D}} where D =
-    symbolization.binning.N^D
-total_outcomes(x::AbstractVector, symbolization::RBE{B, T}) where {B <: FRB, T <: Number} =
-    symbolization.binning.N
-total_outcomes(x::AbstractDataset{D}, symbolization::RBE{B, T}) where {B <: FRB, T <: SVector{D}} where D =
-    symbolization.binning.N^D
+function total_outcomes(x::Array_or_Dataset, b::FixedRectangularBinning)
+    return total_outcomes(RectangularBinEncoding(x, b))
+end
+# This function does not need `x`; all info about binning are in the encoding
+function total_outcomes(e::RectangularBinEncoding)
+    D = length(e.mini)
+    return e.binning.N^D
+end
+
+# This function does not need `x`; all info about binning are in the encoding
+function all_possible_outcomes(e::RectangularBinEncoding)
+    # We can be smart here. All possible bins are exactly the same thing
+    # as the Cartesian Indices of a NxD matrix!
```

Comment:
This is true if the bins are the integer encodings of the bins. Do we have an easy way to convert these `SVector{D, Int}` outcomes to a `SVector{D, Float64}` absolute coordinate?